app/services/VatReturnService.scala

```diff
@@ -39,6 +44,30 @@ class VatReturnService @Inject()() {
         VatReturnRequest(vrn, period, None, None, salesFromNi, salesFromEu)
     )
 
+  def getVatOwedToCountryOnReturn(country: Country, period: Period)(implicit hc: HeaderCarrier, ec: ExecutionContext) = {
+    connector.get(period).map {
+      case Right(vatReturn) =>
+        val sumFromNiToSelectedCountry = vatReturn.salesFromNi.filter(_.countryOfConsumption.code == country.code)
+          .flatMap(x => x.amounts.map(_.vatOnSales.amount)).sum
+        val salesFromEU = vatReturn.salesFromEu.flatMap(_.sales)
+        val sumFromEUToSelectedCountry = salesFromEU.filter(_.countryOfConsumption.code == country.code)
+          .flatMap(x => x.amounts.map(_.vatOnSales.amount)).sum
+        val vatOwedToCountryOnPrevReturn = sumFromEUToSelectedCountry + sumFromNiToSelectedCountry
+        vatOwedToCountryOnPrevReturn
+      case Left(value) =>
+        logger.error(s"there was an error $value")
```

Comment:
can we add some context to the error log as to what's gone wrong

Reply:
done now